shared/strbuf.h

```diff
@@ -17,6 +17,19 @@ struct strbuf {
 	bool heap;
 };
 
+#define DECLARE_STRBUF_WITH_STACK(name__, sz__)   \
+	assert_cc(__builtin_constant_p(sz__));    \
+	char name__##_storage__[sz__];            \
+	_cleanup_strbuf_ struct strbuf name__ = { \
+		.bytes = name__##_storage__,      \
+		.size = sz__,                     \
+	}
+
+#define STRBUF_DECLARE(name__, sz__)            \
```

Comment:
Nits:
 - `s/STRBUF_DECLARE/DECLARE_STRBUF/g` to be consistent with the stack one
 - drop the unused `sz__` to make it obvious it doesn't pre-allocate (stack or heap) anything
 - can use use it across the series?

I greatly prefer these macros over `strbuf_init_with_stack()`.